packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -190,7 +197,9 @@ export function rerenderVM(vm: VM) {
 export function connectRootElement(elm: any) {
     const vm = getAssociatedVM(elm);
 
-    startGlobalMeasure(GlobalMeasurementPhase.HYDRATE, vm);
+    if (profilerEnabled) {
```

Comment:
This pattern is already in place, however, it feels really error-prone to have to wrap all the `logGlobalOperationStart` and `logGlobalOperationEnd` in an `if (profilerEnabled) {}` block. This logic would be better encapsulated if it was part of the `log*Start` and `log*End` operations.

Reply:
@tariqrafique Did you have something in mind with the `profilerEnabled` design? Is it a JS optimization or something? I was kind of wondering the same thing as PM.

Reply:
i found it slowed things a little bit in the 10k+ tests.
the enum lookup is an object property lookup and i wanted to avoid that in the hot path. 

if you can get this to compile with `const enum` which just replaces the enums with constants in the transpiled code, then go for it. 
I couldn't get typescript rollup plugin to work with it a while back. that may have changed.

Reply:
In that case let's leave the current `if (profilerEnabled)` pattern as-is.

As for `const enum`, we can do that in one fell swoop after this PR is merged. We have lots of enums we can convert over.

Reply:
Opened a WI for `const enum`: W-9611979